### PR TITLE
Feat: user message for invalid input

### LINKED ISSUE #36: Create user messages for invalid inputs for Hex functionality
Right now, we have the functionality built out for two hex values. 
If the user provides an invalid input, it just returns NaN.

The goal of this ticket is to display a more helpful message to the user that their input is invalid.
Screen Shot 2022-08-08 at 6 43 30 PM

Get creative with the messaging. 
I think it would help to show the correct format too. 
You can also look to other apps on how they handle invalid inputs

### PR DESCRIPTION
**Pull request by Okibe00**
# Description
Added a message  which displays the supported formats when a users provides an unsupported format.

closes #(36)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Chore items (this includes basic clean up of files, or updates to packages)
- [ ] Updates to documentation

## Checklist

- [x] I have a descriptive title for my PR
- [x] I have linked this PR to the corresponding issue. [See how to do that here.](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [x] I have run the project locally and reviewed the code changes
- [x] My changes generate no new warnings

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -54,6 +54,17 @@ <h2 class="sr-only">Results</h2>
                 <h3 class="text-small">Contrast Ratio</h3>
                 <p id="contrast-ratio-result" class="large"></p>
               </div>
+              <div class="error-message">
+                <h2>
+                  Invalid input
+                </h2>
+                <p>The following input formats are supported
```
Comment (AndreyDmitriev39r) [inline index.html:61]: If your intention was to list all supported formats, then maybe RGB format is needed here?

**Review by AndreyDmitriev39r — CHANGES_REQUESTED** [review]
Here is the thing:
what if user in the process of typing right values? Like below:

17 08 2022_22 10 57_REC

17 08 2022_22 09 17_REC

I think we don't need to remind about valid input right away.
Though I am not sure how achieve this with a single regexp. Maybe we need several regexps for different cases here.

Also we definitely need to find a way _to not display `NaN` here_, I think it was a part of the ticket.

As I see it now, task to create logic for every case of invalid input is a bit extensive. It's cool that you tried that, ahd this regexp looks like a lot of work. However, maybe for the start it would be better to concentrate only on invalid HEX format?

So, to wrap up all this: 

- definitely need to avoid `NaN` output.
- some logic needed in order to display error message not too early. to implement this for all formats at once - could be hard
- thus, I suggest to focus on HEX for this ticket, but it's up to you
- if you need assistance, let me know, I will try to play around your branch more

File: index.html (hunk 1 above)
Comment (Okibe00) [reply index.html:61]: It definitely felt like a jumped the gun when I saw the new issues opened by Jessica.
>some logic needed in order to display error message not too early. to implement this for all formats at once - could be hard
yep that's a Tricky one, already did battle with it making this issue but thinking about it now with a single color input in mind I think I got just the thing for it.
I will get everything and push hopefully before the day runs out.

File: index.js [hunk 2]
```diff
@@ -101,10 +101,24 @@ const displayResult = () => {
...
   }
+  if(firstColor.length >= 7 && secondColor.length >= 7) {
+    if(!hexRegex.test(firstColor) || !hexRegex.test(secondColor)){
+       ratioResult.innerHTML = `
+        <div class = "error-message">
+          <h3>Invalid Input</h3>  
+          <p>The following format is supported. Hexadecimal (e.g #000000, #f1f1f1)</p>
+        </div>`;
+    };
+  }else if(firstColor.length < 7 || secondColor.length < 7) {
```
Comment (AndreyDmitriev39r) [inline index.js:116]: As was said in discord, the check
```
else if(firstColor.length < 7 || secondColor.length < 7)
```
may be not needed

if we modify the return statement in calculatRatio function like this:
```
return ((lighterLum + 0.05) / (darkerLum + 0.05)).toFixed(2) === NaN
                    ? "" : ((lighterLum + 0.05) / (darkerLum + 0.05)).toFixed(2);
```
This is suggestion not change request. Be aware, I just modified it quick and dirty. Additional variable may be appropriate here for readability.
Anybody who will work on #76 should keep this in mind I guess.

**Review by AndreyDmitriev39r — APPROVED** [review]
LGTM!

**Review by sparkO7 — APPROVED** [review]
LGTM!

**Pull request merged**